glustercli/cli/utils.py

```diff
@@ -96,11 +97,47 @@ def set_gluster_socket(path):
     GLUSTERD_SOCKET = path
 
 
+def check_for_xml_errors(data):
+    stdout = data[0]
+    stderr = data[1]
+
+    # depending on the gluster sub-command that's run
+    # it can have a returncode of 0 (meaning success)
+    # however this could mean that the formatting of
+    # the xml was successful and not the command that
+    # was run. We need to check stdout and/or stderr
+    # for the `opRet` xml element and if it's -1, then
+    # format the error accordingly and raise
+    # GlusterCmdException
+
+    # for reasons unknown, some commands will fail and
+    # return to stdout instead of stderr and vice versa
+    error = stdout if stdout else stderr or None
+    if error is not None:
+        try:
+            error = ET.fromstring(error)
+        except Exception:
+            # means parsing xml data failed
+            # so play it safe and ignore
+            pass
```

Comment:
instead of `pass` use `return` so that it will not fail at line 124 on exception

Reply:
noticed your change now. ignore my previous comment

Reply:
Yeah, forgot to push that change when making the called method raise exception. Is fixed now.